src/main/kotlin/no/nav/integrasjon/api/v1/Streams.kt

```diff
@@ -0,0 +1,88 @@
+package no.nav.integrasjon.api.v1
+
+import io.ktor.application.call
+import io.ktor.http.HttpStatusCode
+import io.ktor.locations.Location
+import io.ktor.response.respond
+import io.ktor.routing.Routing
+import no.nav.integrasjon.FasitProperties
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.BasicAuthSecurity
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.Group
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.badRequest
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.ok
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.post
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.securityAndReponds
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.serviceUnavailable
+import no.nav.integrasjon.api.nielsfalk.ktor.swagger.unAuthorized
+import org.apache.kafka.clients.admin.AdminClient
+import org.apache.kafka.common.acl.AccessControlEntry
+import org.apache.kafka.common.acl.AclBinding
+import org.apache.kafka.common.acl.AclOperation
+import org.apache.kafka.common.acl.AclPermissionType
+import org.apache.kafka.common.resource.PatternType
+import org.apache.kafka.common.resource.ResourcePattern
+import org.apache.kafka.common.resource.ResourceType
+import java.util.concurrent.TimeUnit
+
+private const val swGroup = "Streams"
+
+enum class PostStreamStatus {
+    ERROR,
+    OK
+}
+
+@Group(swGroup)
+@Location("$STREAMS/")
+class PostStream
+data class PostStreamBody(val applicationName: String, val user: String)
+
+data class PostStreamResponse(val status: PostStreamStatus, val message: String)
+
+fun Routing.streamsAPI(adminClient: AdminClient?, fasitConfig: FasitProperties) {
+    post<PostStream, PostStreamBody>(
+            "new streams app. The stream app will get permissions to create new internal topics."
+                    .securityAndReponds(
+                            BasicAuthSecurity(),
+                            ok<PostStreamResponse>(),
+                            serviceUnavailable<AnError>(),
+                            badRequest<AnError>(),
+                            unAuthorized<Unit>()
+                    )
+    ) { _, body ->
+        adminClient?.listTopics()?.names()?.get(fasitConfig.kafkaTimeout, TimeUnit.MILLISECONDS)?.filter {
+            it.startsWith(body.applicationName)
```

Comment:
Won't this make it impossible to do changes after its first created?

Reply:
changes to the topic or the stream app name?